azure/scope/machine.go

```diff
@@ -805,3 +806,9 @@ func (m *MachineScope) UpdatePatchStatus(condition clusterv1.ConditionType, serv
 		conditions.MarkFalse(m.AzureMachine, condition, infrav1.FailedReason, clusterv1.ConditionSeverityError, "%s failed to update. err: %s", service, err.Error())
 	}
 }
+
+// GetCapacityReservationGroupID returns the CapacityReservationGroupID from the spec if the
+// value is assigned, or else returns an empty string.
+func (m *MachineScope) GetCapacityReservationGroupID() string {
+	return ptr.Deref(m.AzureMachine.Spec.CapacityReservationGroupID, "")
```

Comment:
Q: what's the benefit of making `CapacityReservationGroupID` a pointer? It seems like the empty string would suffice as an empty value instead of `nil` if we're setting it to an empty string anyway.

Reply:
I used `ProviderID` as example to set the type of `CapacityReservationGroupID` as a string pointer. However, I agree that here we don't differentiate between a nil value and an empty string. PLMK if I should change the type to string.